### DISCUSSION TITLE
Energy and Force RMSE goes up with iteration

### DISCUSSION THREAD

**Discussion by changxiaoju** [deepmodeling/dpgen#557] [Q&A]
system:H
init data: generated from init_bulk, 2000 configurations
training:
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=zh-CN style='font-family:Calibri;font-size:11.0pt'>

<div style='direction:ltr'>

Iter | DPMD | Length[ps] | Ensemble | Temperature[K] | Pressure[Gpa] | Trust   range[ev/A] | Accurate[%] | Candidate[%] | Fail[%]
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
00 | 189 | 0.2=0.0002*1000 | NPT | 200 | 50,100,200,300,400,500,600 | [2.0,2.5] | 25.87 | 13.21 | 60.92
01 | 189 | 0.8=0.0002*4000 | NPT | 200 | 50,100,200,300,400,500,600 | [0.25,0.35] | 40.3 | 33.90 | 25.80
02 | 189 | 4=0.0002*20000 | NPT | 200 | 50,100,200,300,400,500,600 | [0.1,0.125] | 46.7 | 38.11 | 15.19
03 | 189 | 4=0.0002*20000 | NPT | 200 | 50,100,200,300,400,500,600 | [0.1,0.125] | 87.68 | 10.66 | 1.66
04 | 189 | 4=0.0002*20000 | NPT | 200 | 50,100,200,300,400,500,600 | [0.1,0.125] | 82.7 | 3.24 | 14.05
05 | 189 | 8=0.0002*40000 | NPT | 200 | 50,100,200,300,400,500,600 | [0.1,0.125] | 85.21 | 0.44 | 14.35
06 | 189 | 16=0.0002*80000 | NPT | 200 | 50,100,200,300,400,500,600 | [0.1,0.125] | 85.58 | 0.13 | 14.29

</div>

</body>

</html>

but force error goes up with iteration:
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=zh-CN style='font-family:Calibri;font-size:11.0pt'>

<div style='direction:ltr'>

iter | 00 | 01 | 02 | 03 | 04 | 05
-- | -- | -- | -- | -- | -- | --
std(F_DP - F_DFT)(ev/A) | 0.0812 | 0.1323 | 0.1474 | 0.1631 | 0.1629 | 0.1733
std_Force(ev/A) | 0.8658 | 0.8658 | 0.8658 | 0.8658 | 0.8658 | 0.8658
std(F_DF - F_DFT)/std_Force | 0.0938 | 0.1528 | 0.1702 | 0.1884 | 0.1881 | 0.2001

</div>

also can be seen by detail of the error:
iter00:
</body>

![image](https://user-images.githubusercontent.com/44628664/135198569-7339347a-b390-436d-b049-288008d74a5f.png)

iter05:
![image](https://user-images.githubusercontent.com/44628664/135199039-651285bb-259b-4f0c-baaf-19fda64dca01.png)

training error:
iter00:
![image](https://user-images.githubusercontent.com/44628664/135370991-43ba9501-7ee8-477b-8a2c-356af688e1f3.png)
![image](https://user-images.githubusercontent.com/44628664/135371004-9d5ed4aa-7781-448e-9bcd-6adcf4d87dac.png)

iter05:
![image](https://user-images.githubusercontent.com/44628664/135370977-218a1e4b-8bda-4f3d-ac86-6a1602e873b0.png)
![image](https://user-images.githubusercontent.com/44628664/135370961-e50b0299-8d71-4d79-903e-3fa3f4291f72.png)

init data force distribution:
![force](https://user-images.githubusercontent.com/44628664/135374160-195bd21a-e4d8-4060-9a85-0380f5798f48.png)

Energy and Force RMSE goes up. Is this an overfitting problem? It seems that configuration dpgen learned are too close to each other, without H-H bond formation.
Init data generated by dpgen init_bulk
02.md : 100steps*0.2fs=0.02ps,200K,160Gpa
POSCAR:
![image](https://user-images.githubusercontent.com/44628664/135375658-00b83b85-fdfc-4ab0-b7a7-ce127ab2af3a.png)
CONTCAR:
![image](https://user-images.githubusercontent.com/44628664/135376996-52936eff-113c-45f5-98a9-53fcf43e2d77.png)

however, in model_devi, the configuration is like:
![image](https://user-images.githubusercontent.com/44628664/135377190-6aba4d8e-adf1-4ff5-ad63-26a50867a84d.png)

**Comment by jameswind** [comment]
What’s the magnitude of forces? What is the training error? Does increasing
the training steps help?

**Reply by changxiaoju** [reply]
thank you,  i added training error and force distribution above. Increasing steps and "limit_pref_f" did not solve this problem.

**Reply by changxiaoju** [reply]
Maybe something wrong with my origin data?

**Comment by Chenghao-Wu** [comment]
any updates in this issue?
